docs/tutorials/resources/local-ingress-nginx/blueprint/blueprint.yaml

```diff
@@ -0,0 +1,53 @@
+# SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: Blueprint
+
+imports:
+- name: cluster
+  targetType: landscaper.gardener.cloud/kubernetes-cluster
+
+deployExecutions:
+- name: default
+  type: GoTemplate
+  template: |
+    deployItems:
+    - name: deploy
+      type: landscaper.gardener.cloud/helm
+      target:
+        name: {{ .imports.cluster.metadata.name }}
+        namespace: {{ .imports.cluster.metadata.namespace }}
+      config:
+        apiVersion: helm.deployer.landscaper.gardener.cloud/v1alpha1
+        kind: ProviderConfiguration
+
+        chart:
+          fromResource: # will fetch the helm chart from component descriptor resource of type helm chart
```

Comment:
Why is here some new way to specify a resource from a context? In the documentation I could not find any description of it. For me the syntax look quite strange compared to e.g. 
```
{{ $resource := getResource .cd "name" "ingress-nginx-chart" }}
ref: {{ $resource.access.imageReference }}
```

Reply:
your example is valid if the helm chart is a independent helm chart.
In this example, the helm chart is part of the component descriptor. Therefore the component descriptor reference is needed for the helm deployer to fetch the helm chart.

Thats why we need to provide the component descriptor information.
This templating will also be simpler as soon as https://github.com/gardener/landscaper/pull/93 is merged.
Then it will look like:
```
chart:
  fromResource:
{{ toYaml .componentDescriptorDef | indent 6 }}
    resourceName: ingress-nginx-chart
```

Reply:
your example is valid if the helm chart is a independent helm chart.
In this example, the helm chart is part of the component descriptor. Therefore the component descriptor reference is needed for the helm deployer to fetch the helm chart.

Thats why we need to provide the component descriptor information.
This templating will also be simpler as soon as https://github.com/gardener/landscaper/pull/93 is merged.
Then it will look like:
```
chart:
  fromResource:
{{ toYaml .componentDescriptorDef | indent 6 }}
    resourceName: ingress-nginx-chart
```

Reply:
I would expect that the old reference would be sufficient also for local helm charts because in the resource spec of the component descriptor there is the type localOciBlob which signals that the helm chart is local. So the old spec defines which resource entry should be used and the resource entry contains the information how to access this info. 

This makes it even harder for end users to understand this, e.g. what is ". componentDescriptorDef"?

Reply:
The localOCIBlob references the layer but the helm deployer does not know the oci artifact of the layer (which is the component descriptor in this case).

So when such a blob is resolved in the oci you need to provide a context ref (url + version) and then you can fetch the  blobs for that oci artifact.

`componentDescriptorDef` is the component descriptor definition which is also used in the installation.
@MrBatschner maybe we can rename it to be more meaningful.

Reply:
OK, I got it. Not that simple :-). 

Is this already somewhere in the documentation?

Reply:
these local artifacts are described https://github.com/gardener/landscaper/blob/master/docs/concepts/InstallationBlueprintRelationship.md and in this PR's tutorial (https://github.com/gardener/landscaper/blob/bd58ee5598a78b791ba53dcc101c996fc2832d18/docs/images/02-tutorial-local-artifacts.png)

The new component descriptor definition is described in this PR: #93 

Is there anything missing about that?

Reply:
I meant on the level of deploy items in the document [02-local-simple-blueprint.md](https://github.com/schrodit/landscaper/blob/local-bp-tutorial/docs/tutorials/02-local-simple-blueprint.md)

Reply:
ah yes you're right. that is missing  in the tutorial. I will add also a description for the blueprint